src/components/Asset/AssetActions/ButtonBuy/index.test.tsx

```diff
@@ -0,0 +1,291 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import ButtonBuy from './'
+
+//  TESTS FOR DOWNLOAD
+test('Renders Buy button without crashing', async () => {
+  render(
+    <ButtonBuy
+      action="download"
```

Comment:
lots of tests, nice!

They;'re just hard to follow with all those props which is of course caused by this mega component which should simply not have that many props. To keep at least the tests readable would suggest to make use of object spreading, define all the props you will not change as e.g. `defaultProps`, then spread them on component and after it redefine what you are testing, like so for the next `hasPreviousOrder=true` test:

```
<ButtonBuy {...defaultProps} hasPreviousOrder />
```

Reply:
Ok cool, yeah that makes sense. I was thinking the test file was getting a bit too big

Reply:
also the structure should be this, just for consistency's sake, most likely won't change anything during runs:

```js
describe('Asset/AssetActions/ButtonBuy', () => {
  it('Renders Buy button without crashing', () => { ... })
  it('Renders Buy button without crashing when hasPreviousOrder=true', () => { ... })
})
```

Reply:
OK cool, looks good. I'll update it